submodules/k8s/templates/k8s-functions.sh.tftpl

```diff
@@ -4,17 +4,20 @@ RED="\e[31m"
 GREEN="\e[32m"
 EC="\e[0m"
 
+KUBECONFIG="${kubeconfig_path}"
+KUBECONFIG_PROXY="$KUBECONFIG-proxy"
+
 open_ssh_tunnel_to_k8s_api() {
   TUNNEL_SOCKET_FILE=$${TUNNEL_SOCKET_FILE:-/tmp/k8s-tunnel-socket}
   printf "$GREEN Openning k8s tunnel ... $EC \n"
   ssh -q -N -f -M -o "StrictHostKeyChecking no" -o "ExitOnForwardFailure=yes" -i "${ssh_pvt_key_path}" -D ${k8s_tunnel_port} -S "$TUNNEL_SOCKET_FILE" ${bastion_user}@${bastion_public_ip}
 }
 
 check_kubeconfig() {
-  local KUBECONFIG="${kubeconfig_path}"
   printf "$GREEN Checking if $KUBECONFIG exists... $EC \n"
   if test -f "$KUBECONFIG"; then
-    echo "$KUBECONFIG exists." && export KUBECONFIG
+    echo "$KUBECONFIG exists, creating $KUBECONFIG_PROXY for proxy use."
+    sed -e '/server:/a\' -e '    proxy-url: socks5://127.0.0.1:1080' $KUBECONFIG > $KUBECONFIG_PROXY
```

Comment:
Shouldn't the port here (`1080`) actually be a reference to `k8s_tunnel_port`? If not, we should probably make it variable-driven

Reply:
I guess that goes back to the prior discussion on maintaining or minimizing variable use. 🤔

Reply:
Doh, good call.

Reply:
Github didn't show me your second comment when I responded for some reason, but this is just not using the variables we already have like we should, so I just objectively should have used it as you suggested.